Event/OnPasswordResetTokenCreated.php

```diff
@@ -14,11 +15,13 @@ class OnPasswordResetTokenCreated
 
     private $emailFrom;
 
-    public function __construct(Swift_Mailer $mailer, TranslatorInterface $translator, $emailFrom)
+    public function __construct(EventDispatcherInterface $dispatcher, Swift_Mailer $mailer, TranslatorInterface $translator, $emailFrom)
     {
         $this->mailer = $mailer;
         $this->translator = $translator;
         $this->emailFrom = $emailFrom;
+        
+        $dispatcher->addListener(PasswordResetTokenCreated::NAME, [$this, 'onPasswordResetTokenCreated']);
```

Comment:
das hier je event? je event moet ook niets van je listeners weten.
het is de bedoeling dat listeners niet van dispatcher weten en omgekeerd. Het enige wat ze delen is kennis van het event.

Via configuratie (services.yml file) gaat ge de listeners aan het event koppelen.

Er zijn twee soorten in symfony:

* listeners: weten niet naar welk event ze listenen maar krijgen het binnen
* subscribers: subscriben zelf op een event.

Ik heb in de meeste gevallen een voorkeur voor het laatste.